target/scripts/start-mailserver.sh

```diff
@@ -1970,42 +1969,45 @@ function misc
 
 function _misc_save_states
 {
-  # consolidate all states into a single directory (`/var/mail-state`) to allow persistence using docker volumes
-  statedir=/var/mail-state
+  # consolidate all states into a single directory (`/var/mail-state`)
+  # to allow persistence using docker volumes
 
-  if [[ ${ONE_DIR} -eq 1 ]] && [[ -d ${statedir} ]]
+  local STATEDIR=/var/mail-state
+
+  if [[ ${ONE_DIR} -eq 1 ]] && [[ -d ${STATEDIR} ]]
   then
-    _notify 'inf' "Consolidating all state onto ${statedir}"
+    _notify 'inf' "Consolidating all state onto ${STATEDIR}"
 
     local FILES=(
-      /var/spool/postfix
-      /var/lib/postfix
-      /var/lib/amavis
-      /var/lib/clamav
-      /var/lib/spamassassin
-      /var/lib/fail2ban
-      /var/lib/postgrey
-      /var/lib/dovecot
+      spool/postfix
+      lib/postfix
+      lib/amavis
+      lib/clamav
+      lib/spamassassin
+      lib/fail2ban
+      lib/postgrey
+      lib/dovecot
     )
 
-    for d in "${FILES[@]}"
+    for FILE in "${FILES[@]}"
```

Comment:
FILE should be "local"

Reply:
didn't know `for local FILE in ...` was even possible, oder did you mean

``` BASH
local FILE
for FILE in ...
```

which is to the best of my knowledge not needed since `FILE` only exists in the loop. And the FILE I explicitly define in the loop is marked `local`. What did I miss?:)

Reply:
I meant the second example from you.

`FILE` is not declared local before it's first usage in line 1992. So at this point, it's global and keeps the value `spool/postfix`.

If you run e.g. `echo $FILE` after the function call, you would get `spool/postfix`.
```
_misc_save_states
echo $FILE
```

Reply:
I always thought this variable is only valid inside the loop, but I'm not actually surprised that it's not - I guess you can always learn something new.  Bash is still full of surprises.

I will do a follow-up here.